pkg/readiness/ready_tracker.go

```diff
@@ -689,58 +726,87 @@ func (t *Tracker) trackConstraintTemplates(ctx context.Context) error {
 	return nil
 }
 
-// trackConfig sets expectations for cached data as specified by the singleton Config resource.
-// Fails-open if the Config resource cannot be fetched or does not exist.
-func (t *Tracker) trackConfig(ctx context.Context) error {
+// trackSyncSources sets expectations for cached data as specified by the singleton Config resource.
+// and any SyncSet resources present on the cluster.
+// Works best effort and fails-open if the a resource cannot be fetched or does not exist.
+func (t *Tracker) trackSyncSources(ctx context.Context) error {
 	var wg sync.WaitGroup
 	defer func() {
-		defer t.config.ExpectationsDone()
```

Comment:
The point of this defer was to have this line execute after the wg.wait() finished, otherwise we are marking expectations as completed before they are completed.

Reply:
wait, out of curiosity:

shouldn't config and syncset expectations be orthogonal to setting up the data expectations? IOW, why should we wait on the data expectations to be populated in the first place?

Reply:
Good point, gathering configs/syncsets are blocking. In this case we can remove the waitgroup.

But that also means we should be sure that ExpectationsDone() is handled appropriately for the data tracker. Is it?

Reply:
I think so! For each GVK, we set an expectation here:

https://github.com/open-policy-agent/gatekeeper/blob/366ba791d009c70f18f85fd94dbd5aed8426fc9d/pkg/readiness/ready_tracker.go#L732

then go on to mark the populated for the tracker here:

https://github.com/open-policy-agent/gatekeeper/blob/366ba791d009c70f18f85fd94dbd5aed8426fc9d/pkg/readiness/ready_tracker.go#L774

this all look similar to how we do the constraints tracking once they are deduped:

https://github.com/open-policy-agent/gatekeeper/blob/366ba791d009c70f18f85fd94dbd5aed8426fc9d/pkg/readiness/ready_tracker.go#L679-L686